### DISCUSSION TITLE
Make `<script setup>` support `src`

### DISCUSSION THREAD

**Discussion by CarterLi** [vuejs/rfcs#446] [General]
> Due to the difference in module execution semantics, code inside <script setup> relies on the context of an SFC. When moved into external .js or .ts files, it may lead to confusion for both developers and tools. Therefore, <script setup> cannot be used with the src attribute.

If we name the script `MyComponent.vue.ts`  it won't confuse developers

if we explicitly write `import { defineProps, defineEmits, defineExpose } from 'vue'` it won't confuse tools ( at least for IDEs )

I like to move my business logic code into a seperate file and I don't want to make my template file toooooo long. Please consider it twice.

---

Moved from https://github.com/vuejs/core/discussions/5902#discussioncomment-2737232

**Comment by drocha87** [comment]
I'm wondering if it will be to hard to implement a way to infer the code behind without the need to have a `<script setup src="..."/>` (if it have the same name but ends with `.ts` for example `login.vue` and `login.vue.ts`). This is how it works in Blazor for example.

 And now that vscode has [Explorer file nesting](https://code.visualstudio.com/updates/v1_67#_explorer-file-nesting) it's even better.
![image](https://user-images.githubusercontent.com/31552590/168276035-b6b763ea-2ad8-47b3-b792-1551d9ea7645.png)
